### PR TITLE
[html-pie/scatter/bar/linechart] defer to html-chart renderer

### PR DESCRIPTION
**Pull request by christopherlam**
- [x] change old jqplot renderer to call html-chart's renderer instead.
- [x] modify old-chart modifiers and add deprecation warnings

### CODE REVIEW

**Comment by christopherlam** [comment]
fao @gjanssens second commit is the right approach to deprecate old api :)

**Comment by gjanssens** [comment]
So if I understand this code correctly, you have
1. first rewritten the old chart generators to defer actual generation to the new chartjs based api such that no code will actually depend on jqplot any more.
2. written convertors for the various setters from the old api (set width and so on) to do the same in the new api
3. added code to issue a warning to the user in case they still use any of the old api calls

Is that correct ?

File: gnucash/report/report.scm [hunk 1]
```diff
@@ -37,6 +37,21 @@
...
+  ;; this macro applied to old html-bar/line/scatter/pie apis will
+  ;; guard a report writer from passing html-chart objects. this
+  ;; should be removed in 5.x series.
+  (syntax-rules ()
+    ((_ api)
+     (let ((old-api api))
+       (set! api
+         (lambda args
+           (if (and (pair? args) (gnc:html-chart? (car args)))
+               (gnc:error "trying to use old-api on html-chart object.")
```
Comment (gjanssens) [inline gnucash/report/report.scm]: I think a gnc:warning is more appropriate: gnc:error is meant to inform the user the requested command can't be executed at all and hence the expected result won't be achieved. In this case you have a fallback that still does what the user is expecting, but that fallback will eventually go away. And we want to warn the user well in advance.

I'd also try to write the message in a way it is more conductive to a solution. This only tells something is wrong it doesn't help much in making it right. Remember the user/developer may not be very deep into our api as you and I would be.

For example, mention the specific function that was called (if possible), and indicate it's deprecated. And explain in the new api chart properties are set via xyz or point at some ready to use documentation or examples.

File: gnucash/report/html-piechart.scm [hunk 2]
```diff
@@ -53,13 +53,17 @@
...
   (gnc:make-html-piechart-internal '(pixels . -1) '(pixels . -1) #f #f #f #f #f #f #f #f #f #f #f))
 
 (define gnc:html-piechart-data
   (record-accessor <html-piechart> 'data))
 
-(define gnc:html-piechart-set-data!
-  (record-modifier <html-piechart> 'data))
+(define (gnc:html-piechart-set-data! pie newdata)
+  (if (gnc:html-chart? pie)
+      (issue-deprecation-warning "uhhh wrong api.")
```
Comment (gjanssens) [inline gnucash/report/html-piechart.scm]: Uhhh, horrible deprecation warning :(
For all the reasons I have mentioned earlier: this only tells the user they did something wrong, but it won't give them a clue as to how to fix it.

Comment (christopherlam) [reply gnucash/report/html-piechart.scm]: forgot this one

**Comment by christopherlam** [comment]
> * first rewritten the old chart generators to defer actual generation to the new chartjs based api such that no code will actually depend on jqplot any more.
> * written convertors for the various setters from the old api (set width and so on) to do the same in the new api
> * added code to issue a warning to the user in case they still use any of the old api calls

yes, no, yes

The various old-api getters/setters still set old html-bar/line/scatter/pie objects. The old-API renderer will create html-chart object, copy values from old-object into html-chart properties, and call html-chart renderer.

**Pull request merged**